src/Microsoft.ML.Core/ComponentModel/ComponentCatalog.cs

```diff
@@ -264,7 +265,8 @@ public object CreateArguments()
         /// <summary>
         /// A description of a single entry point.
         /// </summary>
-        public sealed class EntryPointInfo
+        [BestFriend]
+        internal sealed class EntryPointInfo
```

Comment:
heads up that if we build entry-point based tooling around ML.Net, it would be convenient to be able to access this, rather than replicate it in his own data models.

Reply:
That may be @sfilipi , however this does not align with the current thinking around entry-points as I understand it, which involves (1) getting the entry-point manifest and then (2) auto-generating an API out of that manifest. Of course, we could imagine that a C# API to entry-points might be something worth having, but I generally prefer to have a solid scenario in mind before I start designing an API for it. Anyway: AFAIK it's our intention that any dependency-injection style stuff *not* be part of the V1 API, unless that has changed in the last week or so.